Update README to be more welcoming (#172)
kubernetes-sigs/node-feature-discovery-operator: README.md

# Node Feature Discovery Operator

The Node Feature Discovery operator is a tool for Kubernetes administrators that makes it easy to detect and understand the hardware features and configurations of a cluster's nodes. With this operator, administrators can easily gather information about their nodes that can be used for scheduling, resource management, and more by controlling the life cycle of the [Node-Feature-Discovery](https://github.com/kubernetes-sigs/node-feature-discovery) DaemonSet.

## How it Works

> The operator works by orchestrating all resources needed to run the Node-Feature-Discovery DaemonSet. This DaemonSet runs on each node in the cluster and detects the features and configurations of the node's hardware. The operator then collects and organizes this information, making it easily accessible.

Reviewer: I'd suggest to drop the term daemonset here as it's nfd-worker that is run as a daemonset and nfd-master that is a deployment

```suggestion
The operator works by orchestrating all resources needed to run the Node-Feature-Discovery (NFD). NFD runs on each node in the cluster and detects the features and configurations of the node's hardware ...
```

```plaintext
The operator then collects and organizes this information, making it easily accessible.
```

What information? Node features(?)  This is a bit confusing. It's NFD which makes them accessible, not the operator.

**nit:** also, I'd suggest to wrap the lines to e.g. 80 characters for easier readability/editability in text editors.

## Quick start

> Get the source code

Reviewer: Do we want to advertise this as the preferred deployment method? How about just

```
kubectl create -f https://operatorhub.io/install/nfd-operator.yaml
```

Author: this is the same `Quick start` from the documentation pages, that way you point at requires an extra step to enable the operatorhub on the user cluster.

I am basically bringing the quick start section from the documentation to the readme for a "quick start"

Author: that's where I am trying to move, I want to have HELM charts for the operator, been busy with other stuff, but I am planning to add HELM soon

> By default it will deploy using the minimal tag image, is
> desired you can simply modify the IMAGE_TAG env var to point to the image
> tag to use.

```bash
IMAGE_TAG={{ site.container_image }}
make deploy
```

By default the operator will watch `NodeFeatureDiscovery` objects
only in the namespace where the operator is deployed in. This is
specified by the `WATCH_NAMESPACE` env variable in the operator
deployment manifest. If unset the operator will watch ALL
namespaces.

Create a NodeFeatureDiscovery instance

```bash
kubectl apply -f config/samples/nfd.kubernetes.io_v1_nodefeaturediscovery.yaml
```

## Documentation

For more detailed information on how to use the Node Feature Discovery operator, please check out our [documentation](https://github.com/kubernetes-sigs/node-feature-discovery-operator/blob/master/docs/index.md)

## Contributing

The Node Feature Discovery operator welcomes contributions, and interested parties are encouraged to take a look at the [contributing guidelines](CONTRIBUTING.md) and [open issues](https://github.com/kubernetes-sigs/node-feature-discovery-operator/issues). We're excited to have you join our community of contributors.

## Support

> If there are any issues or questions about the Node Feature Discovery operator, they can be addressed by opening an issue on the [GitHub repository](https://github.com/kubernetes-sigs/node-feature-discovery-operator/issues) or reaching out on the [Slack channel](https://kubernetes.slack.com/messages/node-feature-discovery-operator).

Reviewer: I don't thing we have a dedicated slack channnel for the operator only

> If there are any issues or questions about the Node Feature Discovery operator, they can be addressed by opening an issue on the [GitHub repository](https://github.com/kubernetes-sigs/node-feature-discovery-operator/issues) or reaching out on the [Slack channel](https://kubernetes.slack.com/messages/node-feature-discovery-operator).

Reviewer: You could point to https://github.com/kubernetes-sigs/node-feature-discovery-operator/issues/new/choose for new issues
